File: converted_source/unoted.lfm.java

import java.awt.*;
import java.awt.event.*;
import javax.swing.*;
import javax.swing.event.*;
import javax.swing.table.*;
// import common.*;

public class NoteEditWindow extends JFrame {
    private static final long serialVersionUID = 1L;
    SpeedButton helpButton;
    JButton OK;
    JButton cancel;
    JTextArea noteEditMemo;
    JCheckBox wrapLines;
    public JPanel mainContentPane;
    public JPanel delphiPanel;
    
    public NoteEditWindow() {
        super();
        this.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE); // HIDE_ON_CLOSE DO_NOTHING_ON_CLOSE
        initialize();
    }
    
    private void initialize() {
        this.setSize(new Dimension(705, 325));
        this.setContentPane(getMainContentPane());
        this.setTitle("Edit note for plant");
    }
    
    private JPanel getMainContentPane() {
        if (mainContentPane == null) {
            mainContentPane = new JPanel();
            mainContentPane.setLayout(new BorderLayout());
            mainContentPane.add(getDelphiPanel(), BorderLayout.CENTER);
        }
        return mainContentPane;
    }
    public void OnClose(ActionEvent event) {
        System.out.println("Closed");
    }
    
    public JPanel getDelphiPanel() {
        if (delphiPanel != null) return delphiPanel;
        Toolkit toolkit = java.awt.Toolkit.getDefaultToolkit();
        delphiPanel = new JPanel();
        delphiPanel.setLayout(null);
        // -- delphiPanel.setLayout(BoxLayout(contentPane, BoxLayout.Y_AXIS));
        // Font buttonFont = new Font("Arial Narrow", Font.PLAIN, 9);
        // Insets buttonMargin = new Insets(1, 1, 1, 1);
        this.setBounds(609, 214, 356, 294  + 80); // extra for title bar and menu
        //  --------------- UNHANDLED ATTRIBUTE: this.OnCreate = FormCreate;
        //  --------------- UNHANDLED ATTRIBUTE: this.OnResize = FormResize;
        //  --------------- UNHANDLED ATTRIBUTE: this.TextHeight = 13;
        //  --------------- UNHANDLED ATTRIBUTE: this.PixelsPerInch = 96;
        //  --------------- UNHANDLED ATTRIBUTE: this.Position = poScreenCenter;
        
        
        Image helpButtonImage = toolkit.createImage("../resources/NoteEditForm_helpButton.png");
        this.helpButton = new SpeedButton("Help", new ImageIcon(helpButtonImage));
        this.helpButton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                helpButtonClick(event);
            }
        });
        this.helpButton.setMnemonic(KeyEvent.VK_H);
        this.helpButton.setBounds(265, 64, 80, 21);
        
        this.OK = new JButton("OK");
        this.OK.setMnemonic(KeyEvent.VK_O);
        this.OK.setBounds(265, 2, 80, 21);
        //  --------------- UNHANDLED ATTRIBUTE: this.OK.ModalResult = 1;
        
        this.cancel = new JButton("Cancel");
        this.cancel.setMnemonic(KeyEvent.VK_C);
        this.cancel.setBounds(265, 25, 80, 21);
        //  --------------- UNHANDLED ATTRIBUTE: this.cancel.ModalResult = 2;
        //  --------------- UNHANDLED ATTRIBUTE: this.cancel.Cancel = True;
        
        this.noteEditMemo = new JTextArea(10, 10);
        this.noteEditMemo.setBounds(2, 2, 259, 261);
        //  --------------- UNHANDLED ATTRIBUTE: this.noteEditMemo.ScrollBars = ssVertical;
        
        this.wrapLines = new JCheckBox("wrap lines");
        this.wrapLines.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                wrapLinesClick(event);
            }
        });
        this.wrapLines.setMnemonic(KeyEvent.VK_W);
        this.wrapLines.setBounds(272, 92, 75, 17);
        
        delphiPanel.add(helpButton);
        delphiPanel.add(OK);
        delphiPanel.add(cancel);
        JScrollPane scroller1 = new JScrollPane();
        scroller1.setBounds(2, 2, 259, 261);;
        scroller1.setViewportView(noteEditMemo);
        delphiPanel.add(scroller1);
        delphiPanel.add(wrapLines);
        return delphiPanel;
    }
    
        
    public void helpButtonClick(ActionEvent event) {
        System.out.println("helpButtonClick");
    }
        
    public void wrapLinesClick(ActionEvent event) {
        System.out.println("wrapLinesClick");
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                NoteEditWindow thisClass = new NoteEditWindow();
                thisClass.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
                thisClass.setVisible(true);
            }
        });
    }

}

File: converted_source/unoted.lfm.py
from common import *

#Useage:
#        import NoteEditWindow
#        window = NoteEditWindow.NoteEditWindow()
#        window.visible = 1

class NoteEditWindow(JFrame):
    def __init__(self, title='Edit note for plant'):
        JFrame.__init__(self, title, windowClosing=self.OnClose)
        self.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE) # HIDE_ON_CLOSE DO_NOTHING_ON_CLOSE
        self.BuildWindow()
        
    def OnClose(self, event):
        print "Closed"
        
    # a menu is top item on the menu bar and can be pulled down for a list of menu items
    def createMenu(self, name, mnemonic=None, menuBar = None):
        menu = JMenu(name)
        if mnemonic:
            menu.setMnemonic(mnemonic)
        if menuBar:
            menuBar.add(menu)
        return menu
        
    def BuildWindow(self):
        toolkit = java.awt.Toolkit.getDefaultToolkit()
        contentPane = self.getContentPane()
        contentPane.setLayout(None)
        # -- contentPane.setLayout(BoxLayout(contentPane, BoxLayout.Y_AXIS))
        buttonFont = Font("Arial Narrow", Font.PLAIN, 9)
        buttonMargin = Insets(1, 1, 1, 1)
        self.setBounds(609, 214, 356, 294  + 80) # extra for title bar and menu
        #  --------------- UNHANDLED ATTRIBUTE: self.OnCreate = FormCreate
        #  --------------- UNHANDLED ATTRIBUTE: self.OnResize = FormResize
        #  --------------- UNHANDLED ATTRIBUTE: self.TextHeight = 13
        #  --------------- UNHANDLED ATTRIBUTE: self.PixelsPerInch = 96
        #  --------------- UNHANDLED ATTRIBUTE: self.Position = poScreenCenter
        
        self.helpButtonImage = toolkit.createImage("../resources/NoteEditForm_helpButton.png")
        self.helpButton = SpeedButton('Help', ImageIcon(self.helpButtonImage), actionPerformed=self.helpButtonClick, font=buttonFont, margin=buttonMargin)
        self.helpButton.setMnemonic(KeyEvent.VK_H)
        self.helpButton.setBounds(265, 64, 80, 21)
        self.OK = JButton('OK', font=buttonFont, margin=buttonMargin)
        self.OK.setMnemonic(KeyEvent.VK_O)
        self.OK.setBounds(265, 2, 80, 21)
        #  --------------- UNHANDLED ATTRIBUTE: self.OK.ModalResult = 1
        self.cancel = JButton('Cancel', font=buttonFont, margin=buttonMargin)
        self.cancel.setMnemonic(KeyEvent.VK_C)
        self.cancel.setBounds(265, 25, 80, 21)
        #  --------------- UNHANDLED ATTRIBUTE: self.cancel.ModalResult = 2
        #  --------------- UNHANDLED ATTRIBUTE: self.cancel.Cancel = True
        self.noteEditMemo = JTextArea(10, 10)
        self.noteEditMemo.setBounds(2, 2, 259, 261)
        #  --------------- UNHANDLED ATTRIBUTE: self.noteEditMemo.ScrollBars = ssVertical
        self.wrapLines = JCheckBox('wrap lines', actionPerformed=self.wrapLinesClick, font=buttonFont, margin=buttonMargin)
        self.wrapLines.setMnemonic(KeyEvent.VK_W)
        self.wrapLines.setBounds(272, 92, 75, 17)
        contentPane.add(self.helpButton)
        contentPane.add(self.OK)
        contentPane.add(self.cancel)
        scroller = JScrollPane(self.noteEditMemo)
        scroller.setBounds(2, 2, 259, 261)
        contentPane.add(scroller)
        contentPane.add(self.wrapLines)
        
    def helpButtonClick(self, event):
        print "helpButtonClick"
        
    def wrapLinesClick(self, event):
        print "wrapLinesClick"
        
if __name__ == "__main__":
    window = NoteEditWindow()
    window.visible = 1
    window.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE)
